lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_sqldatabase
+version_added: "2.5"
+short_description: Manage SQL Database instance
+description:
+    - Create, update and delete instance of SQL Database
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    server_name:
+        description:
+            - The name of the server.
+        required: True
+    name:
+        description:
+            - The name of the database to be operated on (updated or created).
+        required: True
+    tags:
+        description:
+            - Resource tags.
+    location:
+        description:
+            - Resource location.
+    collation:
+        description:
+            - The collation of the database. If createMode is not Default, this value is ignored.
+    create_mode:
+        description:
+            - "Specifies the mode of database creation.\n\nDefault: regular database creation.\n\nCopy: creates a database as a copy of an existing database.
```

Comment:
This really needs to be split up instead of just one long paragraph that seems copied from the Azure Docs. Set each options under the options key and Create a new description category for each option.

```
create_mode:
  description:
  - Specifies the mode of data creation.
  - When C(default) means a regular database creation.
  - When C(copy) will create a database as a copy of an existing data. I(source_database_id) must be specified.
  - When C().....
  default: default
  options:
  - default
  - copy
  - non_readable_secondary
  - online_secondary
  - point_in_time_restore
  - recovery
  - restore
  - restore_long_term_retention_backup
```

I would also convert the options to use snake_cake and not CamelCase as that is what Ansible usually does.